code/game/gamemodes/traitor/traitor.dm

```diff
@@ -52,13 +48,64 @@
 		return FALSE
 	return TRUE
 
-
 /datum/game_mode/traitor/post_setup()
-	for(var/t in pre_traitors)
-		var/datum/mind/traitor = t
-		traitor.add_antag_datum(/datum/antagonist/traitor)
-	..()
+	. = ..()
+
+	for(var/datum/mind/traitor in pre_traitors)
+		var/datum/antagonist/traitor/traitor_datum = new(src)
+		if(ishuman(traitor.current))
+			traitor_datum.delayed_objectives = TRUE
+
+		traitor.add_antag_datum(traitor_datum)
+
+	if(length(pre_traitors))
+		var/random_time = rand(5 MINUTES, 15 MINUTES)
+		addtimer(CALLBACK(src, PROC_REF(late_handout)), random_time)
+
+/datum/game_mode/traitor/proc/traitors_to_add()
+	if(GLOB.configuration.gamemode.traitor_scaling)
+		. = max(1, round((num_players())/(traitor_scaling_coeff)))
+	else
+		. = max(1, min(num_players(), traitors_possible))
+
+/datum/game_mode/traitor/late_handout()
+	var/traitors_to_add = 0
+
+	for(var/datum/mind/traitor_mind as anything in traitors)
+		if(QDELETED(traitor_mind) || !traitor_mind.current) // Explicitly no client check in case you happen to fall SSD when this gets ran
+			traitors_to_add++
+			traitors -= traitor_mind
+			continue
+		for(var/datum/antagonist/traitor/traitor_datum in traitor_mind.antag_datums)
+			traitor_datum.objective_holder.assigned_targets = list()
+			for(var/datum/objective/objective as anything in traitor_datum.objective_holder.objectives)
+				objective.force_reset_target()
+
+			SEND_SOUND(traitor_mind.current, sound('sound/ambience/alarm4.ogg'))
```

Comment:
It's a bit odd to have this in this scope (for each on the antag datums).

Reply:
Moved out of scope!